src/ray/core_worker/experimental_mutable_object_provider.cc

```diff
@@ -109,26 +109,42 @@ void MutableObjectProvider::HandlePushMutableObject(
     RAY_CHECK(it != remote_writer_object_to_local_reader_.end());
     info = it->second;
   }
-  size_t data_size = request.data_size();
-  size_t metadata_size = request.metadata_size();
-
-  // Copy both the data and metadata to a local channel.
-  std::shared_ptr<Buffer> data;
-  const uint8_t *metadata_ptr =
-      reinterpret_cast<const uint8_t *>(request.data().data()) + request.data_size();
-  RAY_CHECK_OK(object_manager_->WriteAcquire(info.local_object_id,
-                                             data_size,
-                                             metadata_ptr,
-                                             metadata_size,
-                                             info.num_readers,
-                                             data));
-  RAY_CHECK(data);
-
-  size_t total_size = data_size + metadata_size;
+  size_t total_data_size = request.total_data_size();
+  size_t total_metadata_size = request.total_metadata_size();
+
+  uint64_t written_so_far = request.written_so_far();
+  uint64_t chunk_size = request.chunk_size();
+
+  std::shared_ptr<Buffer> backing_store;
```

Comment:
Can we improve naming?

Reply:
I changed to object_backing_store for now, though let me know if you have a preference